plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/PartitionFilter.java

```diff
@@ -40,8 +40,8 @@ public PartitionFilter(
             @JsonProperty("partitionKeysFilter") TupleDomain<String> partitionKeysFilter)
     {
         this.hiveTableName = requireNonNull(hiveTableName, "hiveTableName is null");
-        this.partitionColumnNames = partitionColumnNames;
-        this.partitionKeysFilter = TupleDomain.fromColumnDomains(requireNonNull(partitionKeysFilter.getColumnDomains(), "parts is null"));
+        this.partitionColumnNames = requireNonNull(partitionColumnNames, "partitionColumnNames is null");
```

Comment:
unrelated, but why no defensive copy here?

Reply:
i missed that